src/lib.rs

```diff
@@ -221,32 +221,44 @@ fn run_physics_schedule(world: &mut World) {
         .remove_resource::<XpbdLoop>()
         .expect("no xpbd loop resource");
 
-    if xpbd_loop.paused {
-        xpbd_loop.accumulator += DELTA_TIME * xpbd_loop.queued_steps as Scalar;
-        xpbd_loop.queued_steps = 0;
-    } else {
-        let time = world.resource::<Time>();
+    #[cfg(feature = "f32")]
+    let delta_seconds = world.resource::<Time>().delta_seconds();
+    #[cfg(feature = "f64")]
+    let delta_seconds = world.resource::<Time>().delta_seconds_f64();
 
-        #[cfg(feature = "f32")]
-        let delta_time = time.delta_seconds();
+    let time_step = *world.resource::<PhysicsTimestep>();
 
-        #[cfg(feature = "f64")]
-        let delta_time = time.delta_seconds_f64();
+    // Update `DeltaTime` according to the `PhysicsTimestep` configuration
+    let dt = match time_step {
+        PhysicsTimestep::Fixed(fixed_delta_seconds) => fixed_delta_seconds,
+        PhysicsTimestep::Variable { max_dt } => delta_seconds.min(max_dt),
+    };
+    world.resource_mut::<DeltaTime>().0 = dt;
 
-        xpbd_loop.accumulator += delta_time;
+    if xpbd_loop.paused {
+        xpbd_loop.accumulator += dt * xpbd_loop.queued_steps as Scalar;
+        xpbd_loop.queued_steps = 0;
+    } else {
+        xpbd_loop.accumulator += delta_seconds;
     }
 
-    while xpbd_loop.accumulator >= DELTA_TIME {
+    while xpbd_loop.accumulator >= dt && dt != 0.0 {
```

Comment:
```suggestion
    while xpbd_loop.accumulator >= dt && dt > 0.0 {
```
Just to be on the safe side?

Reply:
Yeah I doubt the delta time would ever be negative unless users do something dumb, but it probably is better to use `>` just to be sure